web/src/compose_actions.js

```diff
@@ -436,8 +438,28 @@ export function on_narrow(opts) {
             }
             return;
         }
-        // Do not open compose box if organization has disabled sending
-        // direct messages and recipient is not a bot.
+        // Do not open compose box if sender is not allowed to send direct message.
+        const recipient_ids_string = opts.private_message_recipient
+            .split(",")
+            .map((email) => people.get_by_email(email)?.user_id)
+            .join(",");
+
+        const prevoius_messages_exist = all_messages_data
```

Comment:
previous is misspelled; it's not important functionally but is distracting and makes it harder to review changes. `tools/run-codespell` may help with this sort of thing, but I think we run that in CI, so it must miss this class of error.

Reply:
Fixed.